editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -499,69 +503,89 @@ function drawTargetRectanglesForChildrenOfElement(
   const widthOrHeightComplement = simpleFlexDirection === 'row' ? 'height' : 'width'
 
+  const children = MetadataUtils.getChildrenPaths(metadata, flexElementPath)
+
   interface ElemBounds {
     start: number
     size: number
     end: number
+    index: number
   }
 
-  const pseudoElementBefore: ElemBounds = {
+  const pseudoElementLeftOrTop: ElemBounds = {
     start: parentBounds[leftOrTop],
     size: 0,
     end: parentBounds[leftOrTop],
+    index: forwardsOrBackwards === 'forward' ? -1 : children.length,
   }
-  const pseudoElementAfter: ElemBounds = {
+  const pseudoElementRightOrBottom: ElemBounds = {
     start: parentBounds[leftOrTop] + parentBounds[widthOrHeight],
     size: 0,
     end: parentBounds[leftOrTop] + parentBounds[widthOrHeight],
+    index: forwardsOrBackwards === 'forward' ? children.length : -1,
   }
 
-  const childrenBounds: Array<ElemBounds> = MetadataUtils.getChildrenPaths(
-    metadata,
-    flexElementPath,
-  ).map((childPath) => {
+  const childrenBounds: Array<ElemBounds> = children.map((childPath, index) => {
     const bounds = MetadataUtils.getFrameInCanvasCoords(childPath, metadata)!
     return {
       start: bounds[leftOrTop],
       size: bounds[widthOrHeight],
       end: bounds[leftOrTop] + bounds[widthOrHeight],
+      index: index,
     }
   })
 
   const childrenBoundsAlongAxis: Array<ElemBounds> = [
-    pseudoElementBefore,
+    pseudoElementLeftOrTop,
     ...(forwardsOrBackwards === 'forward' ? childrenBounds : reverse(childrenBounds)),
-    pseudoElementAfter,
+    pseudoElementRightOrBottom,
   ]
 
-  let flexInsertionTargets: Array<CanvasRectangle> = []
+  let flexInsertionTargets: Array<{ rect: CanvasRectangle; insertionIndex: number }> = []
 
   if (targetRectangleSize === 'padded-edge') {
-    for (let index = 0; index < childrenBoundsAlongAxis.length - 1; index++) {
-      const start = childrenBoundsAlongAxis[index].end
-      const end = childrenBoundsAlongAxis[index + 1].start
-
-      const normalizedStart = Math.min(start, end)
-      const normalizedEnd = Math.max(start, end)
+    for (let index = 0; index < childrenBoundsAlongAxis.length; index++) {
+      const bounds = childrenBoundsAlongAxis[index]
 
-      const paddedStart = normalizedStart - ExtraPadding
-      const paddedEnd = normalizedEnd + ExtraPadding
+      const normalizedStart = Math.min(bounds.start, bounds.end)
+      const normalizedEnd = Math.max(bounds.start, bounds.end)
 
       flexInsertionTargets.push(
-        rectFromTwoPoints(
-          {
-            [leftOrTop]: paddedStart,
-            [leftOrTopComplement]: parentBounds[leftOrTopComplement],
-          } as any as CanvasPoint, // TODO improve my type
-          {
-            [leftOrTop]: paddedEnd,
-            [leftOrTopComplement]:
-              parentBounds[leftOrTopComplement] + parentBounds[widthOrHeightComplement],
-          } as any as CanvasPoint, // TODO improve my type
-        ),
+        {
```

Comment:
this is the meat of the fix, instead of creating a single box that goes from "end of preceding sibling" to "start of succeeding sibling", we create one-one fixed sized boxes at the two ends